crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -607,6 +614,39 @@ impl<M: NetworkMsg, K: SignatureKey + 'static> ConnectedNetwork<M, K> for Libp2p
                 .map_err(|_| NetworkError::ShutDown)?;
         }
 
+        // TODO maybe we should lift the metrics mutex up a level and copy the inner pattern
+        // ask during pair programming
+        // or maybe channels would be better?
+        let metrics = self.inner.metrics.clone();
+        if let Some(ref config) = &self.inner.reliability_config {
```

Comment:
nit: perhaps we can annotate this block with something like `#[cfg(feature = "hotshot-testing")]` so that we don't do the check for each message on non-test releases.

Not sure if we even make use of that now, curious on your opinion. I planned to use it in another PR

Reply:
+1 +1 great idea! I'll go ahead and annotate with this